include/alpaka/kernel/TaskKernelGpuUniformCudaHipRt.hpp

```diff
@@ -143,52 +143,129 @@ namespace alpaka
         std::tuple<remove_restrict_t<std::decay_t<TArgs>>...> m_args;
     };
 
+    //! The GPU CUDA/HIP accelerator execution task for non-cooperative kernel.
+    template<typename TApi, typename TAcc, typename TDim, typename TIdx, typename TKernelFnObj, typename... TArgs>
+    class TaskKernelNonCooperativeGpuUniformCudaHipRt final
+        : public TaskKernelGpuUniformCudaHipRt<TApi, TAcc, TDim, TIdx, TKernelFnObj, TArgs...>
```

Comment:
what are the advantages of creating a new class `TaskKernelNonCooperativeGpuUniformCudaHipRt` that inherits from `TaskKernelGpuUniformCudaHipRt` ?

Reply:
To make it symmetrical with `TaskKernelCooperativeGpuUniformCudaHipRt`, which also inherits from `TaskKernelGpuUniformCudaHipRt`

Reply:
But then it leads to a lot of code duplication, for no additional purpose :-(

Thinking about it, is inheritance the best approach here ?
Would you consider adding a template parameter to the task, _e.g._ `bool TCooperative`, and use it with `if constexpr` to distinguish how to submit the kernel ?

Reply:
As a way of example, I've done something similar in #2311 .